asynciterator.ts

```diff
@@ -1220,6 +1220,18 @@ export class SimpleTransformIterator<S, D = S> extends TransformIterator<S, D> {
                        TransformOptions<S, D> & ((item: S, done: () => void) => void)) {
     super(source, options as TransformIteratorOptions<S>);
 
+    if (typeof options === 'object') {
+      let removeOptions = true;
+      for (const key in options) {
+        if (key !== 'autoStart') {
+          removeOptions = false;
+          break;
+        }
+      }
+      if (removeOptions)
+        options = undefined;
+    }
+
     // Set transformation steps from the options
     options = options || (!isSourceExpression(source) ? source : null as any);
```

Comment:
This was the only cause of some problems in the testing - since passing `option = { autoStart: true }` means that the source is not used even if it is not a source expression.

I've got around this by setting the options to undefined if the only key is `autoStart` - but I'm thinking that the tests should just be changed instead

Reply:
I'm sorry, I've been looking at this for 15 minutes now but still don't understand.

So the logic of the `SimpleTransformIterator` is that both of its parameters (`source` and `options`) are optional.

So what the original line 1224 is doing is determining whether `options` is the first or the second argument (given that `source` is optional, it might change position). The easy case is when `options` is non-nullish: then `options` is the second argument. If `options` is nullish, then `options` is the first argument—if that first argument is not a source.

None of that should change, regardless of the `autoStart` default. So probably a test change indeed.

Reply:
Makes sense - will do a test change then :)